pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,385 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+from __future__ import print_function
+
+import numpy as np
+import scipy.constants
+from pyiron.atomistics.master.parallel import AtomisticParallelMaster
+from pyiron_base import JobGenerator
+from sklearn.linear_model import LinearRegression
+from collections import defaultdict
+import warnings
+import itertools
+
+__author__ = "Joerg Neugebauer, Jan Janssen"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "1.0"
+__maintainer__ = "Jan Janssen"
+__email__ = "janssen@mpie.de"
+__status__ = "production"
+__date__ = "Oct 21, 2020"
+
+
+eV_div_A3_to_GPa = (
+    1e21 / scipy.constants.physical_constants["joule-electron volt relationship"][0]
+)
+
+def _fit_coeffs_with_stress(
+        strain,
+        stress,
+        rotations,
+        max_polynomial_order=None,
+        fit_first_order=False,
+    ):
+    higher_terms = _get_higher_order_terms(
+        strain,
+        max_polynomial_order=max_polynomial_order,
+        rotations=rotations,
+        derivative=True,
+    )
+    strain = np.einsum('nik,mkl,njl->nmij',
+                       rotations, strain, rotations).reshape(-1, 9)
+    strain = strain[:, [0, 4, 8, 5, 2, 1]]
+    strain[:,3:] *= 2
+    if higher_terms is not None:
+        strain = np.concatenate((strain, higher_terms), axis=-1)
+    stress = np.einsum('nik,mkl,njl->nmij',
+                       rotations, stress, rotations).reshape(-1, 9)
+    stress = stress[:, [0, 4, 8, 5, 2, 1]]
+    score = []
+    coeff = []
+    for ii in range(6):
+        strain_tmp = strain.copy()
+        strain_tmp[:, 6:] *= strain[:, ii, np.newaxis]
+        reg = LinearRegression(
+            fit_intercept=fit_first_order).fit(strain_tmp, stress[:,ii])
```

Comment:
Nit: line break is hard to read. Do you really go over the limit here? If you need to break it I'd rather break it a second time so the argument is on its own, but I'm open to alternative style interpretations.